### DISCUSSION TITLE
Generate qualities by sources

### DISCUSSION THREAD

**Discussion by xning-lci** [vidstack/player#1356] [Q&A]
Hello guys,

I have a question about the qualities. Like the following sources, I have the same size but different type, it will show two `1920` and `576` as a result. The types are because I want to support all the browsers. Any idea for this?

`[
    {
      src: 'https://flphmoegb.filerobot.com/LaSalle-Vancouver/Programs_Learning-Experiences/Fashion/Fashion-Design/LaSalle_04_Fashion_v2_5_FINAL.mp4?vh=f5ab53&format=webm&width=1920&height=1920',
      type: 'video/webm',
      width: 1920,
      height: 1920,
    },
    {
      src: 'https://flphmoegb.filerobot.com/LaSalle-Vancouver/Programs_Learning-Experiences/Fashion/Fashion-Design/LaSalle_04_Fashion_v2_5_FINAL.mp4?vh=f5ab53&format=mp4&width=1920',
      type: 'video/mp4',
      width: 1920,
      height: 1920,
    },
    {
      src: 'https://flphmoegb.filerobot.com/LaSalle-Vancouver/Programs_Learning-Experiences/Fashion/Fashion-Design/LaSalle_04_Fashion_v2_5_FINAL.mp4?vh=f5ab53&format=webm&width=576&height=576',
      type: 'video/webm',
      width: 576,
      height: 576,
    },
    {
      src: 'https://flphmoegb.filerobot.com/LaSalle-Vancouver/Programs_Learning-Experiences/Fashion/Fashion-Design/LaSalle_04_Fashion_v2_5_FINAL.mp4?vh=f5ab53&format=mp4&width=576',
      type: 'video/mp4',
      width: 576,
      height: 576,
    },
  ]`
  
  
  
![image](https://github.com/user-attachments/assets/242c9c58-93f9-43f7-8e57-9d6a24d44532)

**Comment by mihar-22** [comment]
We don't support multiple formats for source qualities at the moment. It's showing up because we support multiple bitrates. In  general, this isn't really the right way to approach multiple resolutions. Strongly recommend looking into adaptive streaming via HLS as mentioned in our docs.
